Fix storage root and filename formatting to sync Cloudinary uploads and deletes

sethkaufee opened the pull request:
When the `@directus/storage-driver-cloudinary` is enabled, file deletes are not being reflected and the "root" folder configuration will break.

The Cloudinary API has undergone changes with regard to formatting/handling of the Public ID (filename) and folders in API requests. This PR is to address those changes and to fix #18146 and fix #19219.

sethkaufee committed:
fix(storage-driver-cloudinary): Add file extension in list response

Append file extension from Cloudinary query response for image and video assets on "list" calls.

sethkaufee committed:
fix(storage-driver-cloudinary): Replace upload folder parameters

Replace "folder" param with "asset_folder" and "use_asset_folder_as_public_id_prefix" in uploads

rijkvanzanten commented:
@sethkaufee Changes LGTM! Thanks 🙂 Mind doing this step? https://github.com/directus/directus/pull/19231#issuecomment-1648399002 nonsense legal requirement for bigger COSS projects

sethkaufee committed:
fix(storage-driver-cloudinary): Fix asset folder params and non media uploads

Only pass "asset_folder" in upload if there is a "root" path configured. For non image and video assets, parse the base path for the public id.

sethkaufee committed:
chore(contributors): added contributor

added self to contributors.yml

sethkaufee commented:
@rijkvanzanten, I tested this with images, video, and other file types, and it seems to be working now with and without a storage "root" configured. 

However, I did run into an `Inconsistent chunk` error when uploading a large videos (66Mb and 15Mb) in both the released version of the driver and with our changes:
```log
[08:44:19.872] WARN: Can't upload file "be52863c-1fa9-45c8-9a06-9e79f5710aeb.mov": Inconsistent chunk sizes
    err: {
      "type": "Error",
      "message": "Can't upload file \"be52863c-1fa9-45c8-9a06-9e79f5710aeb.mov\": Inconsistent chunk sizes: Inconsistent chunk sizes",
      "stack":
          Error: Can't upload file "be52863c-1fa9-45c8-9a06-9e79f5710aeb.mov": Inconsistent chunk sizes
              at DriverCloudinary.write (file:///directus/node_modules/.pnpm/@directus+storage-driver-cloudinary@10.0.7/node_modules/@directus/storage-driver-cloudinary/dist/index.js:234:19)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async FilesService.uploadOne (file:///directus/node_modules/.pnpm/@directus+api@12.0.3_@types+node@20.4.4/node_modules/@directus/api/dist/services/files.js:64:13)
              at async Multipart.<anonymous> (file:///directus/node_modules/.pnpm/@directus+api@12.0.3_@types+node@20.4.4/node_modules/@directus/api/dist/controllers/files.js:93:32)
          caused by: Error: Inconsistent chunk sizes
              at DriverCloudinary.uploadChunk (file:///directus/node_modules/.pnpm/@directus+storage-driver-cloudinary@10.0.7/node_modules/@directus/storage-driver-cloudinary/dist/index.js:254:19)
              at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
              at async file:///directus/node_modules/.pnpm/p-queue@7.3.4/node_modules/p-queue/dist/index.js:118:36
    }
[08:44:20] POST /files 503 6.4s
```

I did not look into this problem in detail, but I can attempt to resolve this as well as I will be needing to upload large video assets for my project. Should I open a separate issue/PR? Thanks!

rijkvanzanten commented:
> I did not look into this problem in detail, but I can attempt to resolve this as well as I will be needing to upload large video assets for my project. Should I open a separate issue/PR? Thanks!

Lets handle this as a separate issue/PR yeah 🙂 We're pushing out a release today, so I'd love for this first fix to be included now 🚀
